rawdb/rawdb.go

```diff
@@ -78,6 +78,9 @@ func (r *RawDB) Close() error {
 	log.Tracef("Close")
 	defer log.Tracef("Close exit")
 
+	r.mtx.Lock()
```

Comment:
This should not need a mutex.

Reply:
@marcopeereboom I believe it does, previously this data race would be detected sometimes, am I misreading this? 🤔 
```
2024-12-04T20:48:00.4826202Z 2024-12-04 20:47:49 INFO tbc tbc.go:2209 RPC server shutdown cleanly
2024-12-04T20:48:00.4826484Z ==================
2024-12-04T20:48:00.4826669Z WARNING: DATA RACE
2024-12-04T20:48:00.4826878Z Read at 0x00c000323e00 by goroutine 337:
2024-12-04T20:48:00.4827267Z   github.com/syndtr/goleveldb/leveldb/storage.(*fileStorage).doLog()
2024-12-04T20:48:00.4828110Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/storage/file_storage.go:187 +0x84
2024-12-04T20:48:00.4828774Z   github.com/syndtr/goleveldb/leveldb/storage.(*fileStorage).log()
2024-12-04T20:48:00.4829596Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/storage/file_storage.go:236 +0x66
2024-12-04T20:48:00.4830375Z   github.com/syndtr/goleveldb/leveldb/storage.(*fileWrap).Sync()
2024-12-04T20:48:00.4831195Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/storage/file_storage.go:592 +0x155
2024-12-04T20:48:00.4831840Z   github.com/syndtr/goleveldb/leveldb.(*iStorageWriter).Sync()
2024-12-04T20:48:00.4832193Z       <autogenerated>:1 +0x43
2024-12-04T20:48:00.4832518Z   github.com/syndtr/goleveldb/leveldb.(*session).flushManifest()
2024-12-04T20:48:00.4833405Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/session_util.go:482 +0x145
2024-12-04T20:48:00.4834008Z   github.com/syndtr/goleveldb/leveldb.(*session).commit()
2024-12-04T20:48:00.4834735Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/session.go:230 +0x1c5
2024-12-04T20:48:00.4835362Z   github.com/syndtr/goleveldb/leveldb.(*DB).compactionCommit.func1()
2024-12-04T20:48:00.4836149Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/db_compaction.go:264 +0x54
2024-12-04T20:48:00.4836798Z   github.com/syndtr/goleveldb/leveldb.(*compactionTransactFunc).run()
2024-12-04T20:48:00.4837594Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/db_compaction.go:242 +0x42
2024-12-04T20:48:00.4838217Z   github.com/syndtr/goleveldb/leveldb.(*DB).compactionTransact()
2024-12-04T20:48:00.4839006Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/db_compaction.go:186 +0x35b
2024-12-04T20:48:00.4839651Z   github.com/syndtr/goleveldb/leveldb.(*DB).compactionTransactFunc()
2024-12-04T20:48:00.4840441Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/db_compaction.go:253 +0x244
2024-12-04T20:48:00.4841064Z   github.com/syndtr/goleveldb/leveldb.(*DB).compactionCommit()
2024-12-04T20:48:00.4842126Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/db_compaction.go:263 +0xc8
2024-12-04T20:48:00.4842758Z   github.com/syndtr/goleveldb/leveldb.(*DB).tableCompaction()
2024-12-04T20:48:00.4843559Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/db_compaction.go:584 +0x14d7
2024-12-04T20:48:00.4844204Z   github.com/syndtr/goleveldb/leveldb.(*DB).tableAutoCompaction()
2024-12-04T20:48:00.4844986Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/db_compaction.go:644 +0x53
2024-12-04T20:48:00.4845624Z   github.com/syndtr/goleveldb/leveldb.(*DB).tCompaction()
2024-12-04T20:48:00.4846380Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/db_compaction.go:863 +0x6f2
2024-12-04T20:48:00.4846980Z   github.com/syndtr/goleveldb/leveldb.openDB.gowrap3()
2024-12-04T20:48:00.4847675Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/db.go:155 +0x33
2024-12-04T20:48:00.4848069Z 
2024-12-04T20:48:00.4848203Z Previous write at 0x00c000323e00 by goroutine 297:
2024-12-04T20:48:00.4848605Z   github.com/syndtr/goleveldb/leveldb/storage.(*fileStorage).doLog()
2024-12-04T20:48:00.4849449Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/storage/file_storage.go:219 +0x1207
2024-12-04T20:48:00.4850122Z   github.com/syndtr/goleveldb/leveldb/storage.(*fileStorage).Log()
2024-12-04T20:48:00.4850949Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/storage/file_storage.go:230 +0x133
2024-12-04T20:48:00.4851562Z   github.com/syndtr/goleveldb/leveldb.(*session).log()
2024-12-04T20:48:00.4852293Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/session_util.go:33 +0x98
2024-12-04T20:48:00.4853034Z   github.com/syndtr/goleveldb/leveldb.(*DB).log()
2024-12-04T20:48:00.4853739Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/db_util.go:37 +0xf3
2024-12-04T20:48:00.4854283Z   github.com/syndtr/goleveldb/leveldb.(*DB).Close()
2024-12-04T20:48:00.4854968Z       /home/runner/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.1-0.20210819022825-2ae1ddf74ef7/leveldb/db.go:1148 +0x71
2024-12-04T20:48:00.4855702Z   github.com/hemilabs/heminetwork/database/level.(*Database).Close()
2024-12-04T20:48:00.4856301Z       /home/runner/work/heminetwork/heminetwork/database/level/level.go:81 +0x6a4
2024-12-04T20:48:00.4856801Z   github.com/hemilabs/heminetwork/database/tbcd/level.(*ldb).Close()
2024-12-04T20:48:00.4857177Z       <autogenerated>:1 +0x46
```